src/etc/inc/util.inc

```diff
@@ -1811,23 +1811,29 @@ function alias_expand_urltable($name) {
 	return null;
 }
 
-/* obtain MAC address given an IP address by looking at the ARP table */
-function arp_get_mac_by_ip($ip) {
-	mwexec("/sbin/ping -c 1 -t 1 " . escapeshellarg($ip), true);
-	$arpoutput = "";
-	exec("/usr/sbin/arp -n " . escapeshellarg($ip), $arpoutput);
-
-	if ($arpoutput[0]) {
-		$arpi = explode(" ", $arpoutput[0]);
-		$macaddr = $arpi[3];
-		if (is_macaddr($macaddr)) {
-			return $macaddr;
-		} else {
-			return false;
-		}
+/* obtain MAC address given an IP address by looking at the ARP/NDP table */
+function arp_get_mac_by_ip($ip, $do_ping = false) {
```

Comment:
I prefer to keep the old behavior as default just to avoid having surprises on edge cases. Please change default $do_ping value to true.